src/Microsoft.Sbom.Api/Config/ConfigSanitizer.cs

```diff
@@ -115,6 +118,26 @@ private ConfigurationSetting<string> GetNamespaceBaseUriFromAssembly(Configurati
             };
         }
 
+        private ConfigurationSetting<string> GetPackageSupplierFromAssembly(Configuration configuration)
+        {
+            if (string.IsNullOrWhiteSpace(assemblyConfig.DefaultPackageSupplier))
+            {
+                return configuration.PackageSupplier;
+            }
+
+            // Give priority to package supplier provided as an argument.
+            if (!string.IsNullOrWhiteSpace(configuration.PackageSupplier?.Value))
```

Comment:
The default value should have higher priority, can you also print a warning as we do in the getnamespaceuri method? We want to let folks know that if a default value is present, providing this argument value will be useless and we will use the default value anyway. That way they won't get confused when this param doesnt work as expected.

Reply:
Actually your suggestion appears counterintuitive to me :) Usually it is expected to supply default value if nothing is provided and use provided value if exists. The behavior where default value has higher priority actually bring confusion (and require to provide a warning so user understand what's going on) and also reduces flexibility, e.g. what if you need to override default value via parameter (example case would be all orgs in MSFT use MSFT, but LinkedIn would like to use LinkedIn as an org value, but still use MSFT-built version).

Reply:
@edgarrs @daneshbadlani thoughts? Should we prioritize default value over supplier or should we provide ability to customize supplier value even when default incorporated into the tool?

Reply:
The 'default' value is a misnomer here. What we want is a mandated value, as in the company you are generating the SBOM for can create a mandated value or aka compliant value that *must* be used to generate the SBOM. 

So, in the normal case, the user *will* have to provide a value here since there is no mandated/default value.

However, a company might want to enforce the name or namespace value such that its developers won't be able to override this value. In such a case, they will add this assembly attribute that will enforce that all SBOMs will have the same company name.

Reply:
I think by providing a parameter value in the CLI or API call, the user expects this value to override any other default\previously set value. So I agree with Anton that the parameter value should have a higher priority than the assembly attribute

Reply:
If that is the case, we don’t even need the assembly attribute. We can just set the default value on the configuration class. The assembly attribute is only there for companies to enforce a particular value.

Reply:
But that means that anyone can override the default name and we will have to catch it in validation and fail so as to enforce they don’t use a non conforming value